### ISSUE TITLE
Robustness issues

### ISSUE THREAD

**Issue by jrmuizel** [fschutt/polyclip#2]
I played around with the reference implementation of this a while ago and ran into a bunch of issues around floating point intersection. From what I remember, the algorithm was getting confused when computed intersections were not showing up in the right order because of rounding errors.

Here's a quick diagram of how things can potentially go wrong:
![intersection](https://user-images.githubusercontent.com/332653/34548352-9ee0ec62-f0cf-11e7-85ff-05141f398ce9.png)
p1 is the rounded intersection of A and D. Even though C is supposed to be a subsegment of A it can end up with a different intersection point with B than A would have.

**Comment by jrmuizel** [comment]
I can try to dig up more about the problem if you're interested.

**Comment by fschutt** [comment]
Yeah, numerical robustness can be an issue. Any information on this is useful, thanks. 

I haven't worked on this library for a while, the only thing missing is porting the commented out part with iterators properly. I can't say if the algorithm will work 100% in the end.
